optimizely/marketing-website
# Upgrade current build to Assemble V6 and add translation key tags

This PR addresses migrating our current build over to Assemble v0.6.0. This is a difficult QA because it effects the templates/yml of every page/layout/partial/modal on the website. The reason for such a wide spread effect is the following:

- [x] scope all yml data locally under the `page_data` namespace
  - [x] scope yml with a `global_` prefix globally
- [x] remove `.hbs` extension from all `layout` YFM references (as new partners are merged in these prefixes may creep in until we deploy.
- [x] change all references to `page.<data_name>` to `this.<data_name>`
- [x] body content injection syntax is switched to `{% body %}`
- [x] partials are referenced {{partial "partial_file_name"}}
- [x] modals are a custom type and no longer being double compiled. Therefore, the `*_compiled*` suffix is omitted, so don't use it in the YFM of pages.
- [ ] watch task that triggers rebuilding assemble is managed by assemble and not grunt. Livereload is still in `grunt watch` (in progress)

#### In Addition:
- [x] Smartling files API and Subfolders build are waiting in the background to be turned on
- [x] TR | MD tags are appended to all external YML and YFM (these are not referenced in .hbs templates because they are being removed before rendering.
- [x] Data is being translated for subfolders (running `grunt server:test` will translate the `website` and not build subfolders.
- [x] Markdown is being parsed on the fly and `{{#omarkdown}}` helpers are no longer needed. Markdown does need triple `{{{ }}}` because it is HTML coming through. 
- [x] Some pages were restructured to account for translation. [See this doc](https://docs.google.com/a/optimizely.com/spreadsheets/d/1sZpLUxlU-pv9ncn0o5_NoDLysgjnZHCwC8TYXBhRpEg/edit?usp=sharing) for an overview

#### To Test

- reference the [Google Sheet](https://docs.google.com/a/optimizely.com/spreadsheets/d/1sZpLUxlU-pv9ncn0o5_NoDLysgjnZHCwC8TYXBhRpEg/edit#gid=0)
- check [Staging](https://www.optimizelystaging.com/dfoxpowell/upgrade-assemble/)
- run all UI Nightmare tests
- David needs to make a doc to divided QA tasks up. 
  - On every page we should be checking all links and forms. Modals that are page dependent should be tested
- UI regressions should be looked for (Missing content, missing buttons, improper alignments, etc). All pages should be compared side by side with their equivalent in production. 
- Pages with partials should be reviewed to see if the partial is included.
- Pages with `global_` yml should be QA'd to see if that content is present
- Pages with external yml should be QA,d to see if content is present
- Try changing content and seeing if `assemble.watch` and `grunt watch:livereload` work
- Run both `grunt server` and `grunt server:test` to see if translated content works
- Check for random `\n` characters in translated pages that may be present from the parser.

## Testing Plan

Everyone:
- [ ] *Layout*: WRAPPER
  - [ ] /index (hompage)

@kylerush 
- [ ] *Layout*: ABOUT
  - [ ] /contact
  - [ ] /press
- [ ] *Layout*: CUSTOM
  - [ ] /resources/live-demo-webinar
- [ ] *Layout*: BENEFITS
  - [ ] /benefits
  - [ ] /ecommerce
  - [ ] /enterprises
  - [ ] /publishers
  - [ ] /small-business
- [ ] *Layout*: OM/ORIGINAL & OM/WRAPPER
  - [ ] /om/free-trial
  - [ ] /om/free-trial/fixture.html
- [ ] *Layout*: CUSTOMERS
  - [ ] /customers/customer-stories
- [ ] *Layout*: FAQ
  - [ ] /faq
  - [ ] /gettingstarted
- [ ] *Layout*: TRIAL_FORM
  - [ ] /free-trial
  - [ ] /resources/ab-testing-tool
  - [ ] /resources/split-testing-tool

@stewsmith 
- [ ] *Layout*: NO-SUB-NAV
  - [ ] /partial/feature-list
  - [ ] /feature-list
- [ ] *Layout*: NO-SUB-NAV-OPTIONAL-HEADER
  - [ ] /features-and-plans
  - [ ] /partial/faq_section
  - [ ] /partial/companies_panel
  - [ ] /partial/features_list
  - [ ] /pricing
- [ ] *Layout*: SEO
  - [ ] /ab-testing 
  - [ ] /split-testing 
  - [ ] /resources/conversion-optimization-tools
  - [ ] /resources/ios-ab-testing
  - [ ] /resources/multivariate-test-vs-ab-test
  - [ ] /resources/multivariate-testing
  - [ ] /resources/online-marketing
  - [ ] /resources/website-optimization
- [ ] *Layout*: RESOURCES
  - [ ] /resources/index (check all individual resources render correctly and are translated)
- [ ] *Layout*: OPTICON
   - [ ] /all opticon pages

@dtothefp
- [ ] *Layout*: PAGE
  - [ ] /communities
  - [ ] /opt_out
  - [ ] /partners/solutions
  - [ ] /partners/technology
  - [ ] /partners/single/index (potentially not being used)
  - [ ] /privacy
  - [ ] /privacy/12/16/2013 (potentially not being used)
  - [ ] /resources/sample-size-calculator
  - [ ] /terms-api
  - [ ] /terms
- [ ] *Layout*: PAGE_SUCCESS
  - [ ] /success
- [ ] *Layout*: STATISTICS
  - [ ] /statistics
- [ ] *Layout*: EVENTS_FORM
  - [ ] /sxsw2015
- [ ] *Layout*: PARTNERS
  - [ ] / all solutions partners pages
  - [ ] / all technology partners pages
- [ ] *Layout*: JOBS
  - [ ] /jobs
- [ ] *Layout*: MOBILE
- [ ] *Layout*: NONPROFITS
  - [ ] /nonprofits
